### PR TITLE
Add analytics-pipeline as a mounted dir.

### PR DESCRIPTION
**Pull request by benpatterson**
This commit does not introduce users; the vagrant user would run tasks.

@mulby @rocha @e0d 

Thoughts?

### CODE REVIEW

File: vagrant/release/devstack/Vagrantfile [hunk 1]
```diff
@@ -109,6 +111,8 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
       create: true, nfs: true
     config.vm.synced_folder "#{ora_mount_dir}", "/edx/app/ora/ora",
       create: true, nfs: true
+    config.vm.synced_folder "#{analytics_pipeline_mount_dir}", "/edx/app/pipeline",
```
Comment (feanil) [inline vagrant/release/devstack/Vagrantfile:116]: Don't forget to add this to the VBOXFS section above as well, or refactor this to something cleaner.

Comment (benpatterson) [inline vagrant/release/devstack/Vagrantfile:116]: Done.

**Comment by benpatterson** [comment]
Addressed @feanil 's comment, but perhaps I should more directly ask:
- Do we consider having a pipeline user a requirement here?
- If so, do we want the requirements to be installed as well? I want to say no since that will be an expensive process that will only be useful to a certain subset of users
- Does the path /edx/app/pipeline make sense? I haven't tested the underlying mkdir functionality and am not sure if it'll give us subdirs for free (e.g., /edx/app/analytics/pipeline)

Thx for any comments here. I hope to pick away at this while I'm doing some dev in this space.

**Comment by feanil** [comment]
I only see the vagrant file changes, are there changes to the vagrant-devstack.yml file as well?  If it is a top-level standalone app then I think it belongs under '{{ COMMON_APP_DIR }}/<app_name>'.  Each top-level app should have it's own user and requirements file.  Is there a PR for the work of that role?

**Comment by rocha** [comment]
@benpatterson what is the goal with this PR? Given that the pipeline will not work without all its dependencies, the only thing that the PR does is bring the code in. Is that a requisite for some of the work you are doing?

**Comment by benpatterson** [comment]
@rocha I suppose the goal of the PR would be to ultimately include the pipeline in devstack, but I'm not sure what the requirements for that would be. I was hoping I could start with this, then hear what else is needed and work its way into something that is mergeable.

Personally, I mounted my dir so I could run jobs on devstack. It'll shorten the cycle a little between creating events and running the job. I am configured to look in /edx/log/tracking. I only need to rename the log file on devstack to be sure it's found. (It will also save me from encountering any OSX-only weirdness).

**Comment by rocha** [comment]
@benpatterson are you able to run the jobs you need without using Hadoop?

**Comment by benpatterson** [comment]
@rocha I use the emulator

**Comment by benpatterson** [comment]
@rocha I had been imagining checking in a devstack.cfg file into the pipeline repo as part of the deal

File: vagrant/release/devstack/Vagrantfile [hunk 2]
```diff
@@ -100,6 +102,8 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
       create: true, owner: "forum", group: "www-data"
     config.vm.synced_folder "#{ora_mount_dir}", "/edx/app/ora/ora",
       create: true, owner: "ora", group: "www-data"
+    config.vm.synced_folder "#{analytics_pipeline_mount_dir}", "/edx/app/pipeline",
+      create: true, owner: "ora", group: "www-data"
```
Comment (brianhw) [inline vagrant/release/devstack/Vagrantfile]: You have the owner here as "ora".  Is that appropriate?

Comment (benpatterson) [inline vagrant/release/devstack/Vagrantfile]: Fixed. Thanks for catching that.

**Comment by e0d** [comment]
@benpatterson @feanil What's the status of this PR?

**Comment by benpatterson** [comment]
I'd need some feedback from the team on how to proceed.

Excepting that, I think we should create another user so a developer can just go sudo su pipeline and land at the new $HOME with an environment activated. If we fully build the virtualenv, it will add some cycles to the initial devstack provisioning. We could also pare this down to have an empty environment that a user would need to create by running the make scripts.

@mulby do you happen to have an opinion here?

**Comment by benpatterson** [comment]
CC @brianhw not sure if you have an opinion on devstacky stuff....

**Comment by rocha** [comment]
@benpatterson is there a way to make the pipeline optional, so it does not have to be included and build into the stack unless specified? The list of requirements for the pipeline can be quite long, and like you mentioned it could add unnecessary cycles to the provisioning when the pipeline is not required.

**Comment by mulby** [comment]
Closing since we now support an analytics devstack

**Pull request closed without merging**